### PR TITLE
Add Snapshot Lifecycle Retention documentation

### PR DESCRIPTION
**Pull request by dakrone**
This commits adds API and general purpose documentation for SLM
retention.

Relates to #43663

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/oss-distro-docs

File: docs/reference/ilm/apis/slm-api.asciidoc [hunk 1]
```diff
@@ -72,6 +78,10 @@ PUT /_slm/policy/daily-snapshots
 <3> Which repository to take the snapshot in
 <4> Any extra snapshot configuration
 <5> Which indices the snapshot should contain
+<6> Optional retention configuration
+<7> Keep snapshots for 30 days
+<8> Always keep at least 5 successful snapshots
+<9> Keep no more than 50 successful snapshots
```
Comment (gwbrown) [inline docs/reference/ilm/apis/slm-api.asciidoc]:
```suggestion
<9> Keep no more than 50 successful snapshots, even if they're less than 30 days old
```

File: docs/reference/ilm/apis/slm-api.asciidoc [hunk 2]
```diff
@@ -72,6 +78,10 @@ PUT /_slm/policy/daily-snapshots
 <3> Which repository to take the snapshot in
 <4> Any extra snapshot configuration
 <5> Which indices the snapshot should contain
+<6> Optional retention configuration
+<7> Keep snapshots for 30 days
+<8> Always keep at least 5 successful snapshots
```
Comment (gwbrown) [inline docs/reference/ilm/apis/slm-api.asciidoc]:
```suggestion
<8> Always keep at least 5 successful snapshots, even if they're more than 30 days old
```

File: docs/reference/ilm/getting-started-slm.asciidoc [hunk 3]
```diff
@@ -105,6 +115,10 @@ PUT /_slm/policy/nightly-snapshots
 <3> the repository the snapshot should be stored in
 <4> the configuration to be used for the snapshot requests (see below)
 <5> which indices should be included in the snapshot, in this case, every index
+<6> Optional retention configuration
+<7> Keep snapshots for 30 days
+<8> Always keep at least 5 successful snapshots
+<9> Keep no more than 50 successful snapshots
```
Comment (gwbrown) [inline docs/reference/ilm/getting-started-slm.asciidoc]: Same comments as above

File: docs/reference/ilm/slm-retention.asciidoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+The cluster level settings for retention are shown below, and can be changed dynamically using the
+<<cluster-update-settings,cluster-update-settings>> API:
+
+|=====================================
+| Setting | Default value | Description
+
+| `slm.retention_schedule` | `0 30 1 * * ?` | A periodic or absolute time schedule for when
+  retention should be run. Supports all values supported by the cron scheduler: <<schedule-cron,Cron
+  scheduler configuration>>. Retention can also be manually run using the
+  <<slm-api-execute-retention,Execute retention API>>. Defaults to daily at 1:30am.
```
Comment (gwbrown) [inline docs/reference/ilm/slm-retention.asciidoc]: We should specify that the default is relative to the server's configured time zone.

**Review by AthenaEryma — APPROVED** [review]
Left a few minor comments, but no need for another round of review after they're addressed.

**Pull request merged**